src/gui/mainwindow.cpp

```diff
@@ -104,8 +110,41 @@ GUI::MainWindow::MainWindow(Pipeline::Pipeline *pipeline, QWidget *parent)
           SLOT(updateVideoFrame(cv::Mat)));
 }
 
+void GUI::MainWindow::updatePostureNotification(
+    PostureEstimating::PostureState postureState) {
+  // Create notification widgets and set the minimum size shown
+  QWidget *postureNotificationBox = new QWidget;
+  postureNotificationBox->setMinimumSize(200, 40);
+  QGridLayout *postureNotificationLayout = new QGridLayout;
+  QLabel *postureNotification = new QLabel();
+
+  // Check if the ideal pose has been set and if so, display notification
+  // according to the posture state
+  if (postureState != 2) {
+    if (postureState == 0) {
+      postureNotificationBox->setStyleSheet("background-color: green");
+      postureNotification->setText("Good Posture");
+    } else if (postureState == 1) {
+      postureNotificationBox->setStyleSheet("background-color: red");
+      postureNotification->setText("Bad Posture");
+    } else {
+      postureNotificationBox->setStyleSheet("background-color: orange");
+      postureNotification->setText("Undefined");
```

Comment:
Should we do this using a case statement? It might make the hole thing a little more readable.